R/utils.R

```
@@ -1,56 +1,102 @@
 #'
-#' @importFrom stringr str_split
-#' @importFrom rlang abort
+#' @importFrom rlang abort inform
 #'
 #' @noRd
-create_item <- function(pkg, type, item_name){
+create_item <- function(pkg, type = c("specification","test_case","test_code"), item_name){
 
-  # Error out if no validation skel
-  if(!dir.exists(paste0(c(pkg, "inst", "validation"),
-                        sep = .Platform$file.sep, collapse = ""))) {
+  type <- match.arg(type)
+
+  validation_directory <- getOption("vt.validation_directory", default = "vignettes/validation")
+
+  # Error out if no validation skeleton
+  if(!dir.exists(file.path(pkg, validation_directory))) {
     abort("No validation structure found. Run `valtools::vt_use_validation().`",
           class = "vt.missingStructure")
 
     # Create item folder if this is the first item
-  } else if(!dir.exists(paste0(c(pkg, "inst", "validation", item),
-                               sep = .Platform$file.sep, collapse = ""))) {
-    dir.create(paste0(c(pkg, "inst", "validation", type),
-                      sep = .Platform$file.sep, collapse = ""),
-               recursive  = TRUE)
   }
 
+  if(!dir.exists(file.path(pkg, validation_directory, type))) {
+    dir.create(file.path(pkg, validation_directory, type))
+  }
+
+
   # Split out item_name to get any directories and create the structure.
-  item_name_split <- str_split(item_name, .Platform$file.sep)[[1]]
-  # Path to new validation item
-  val_dir_path <- paste0(c(pkg, "inst", "validation", type,
-                           paste0(head(item_name_split,-1)),collapse = ""),
-                         sep = .Platform$file.sep, collapse = "")
+  item_dir <- dirname(item_name)
+
   # If the item is nested in a folder make sure its made.
-  if(length(item_name_split) > 1){
+  if( !item_dir %in% c("",".")){
     tryCatch({
-      dir.create(val_dir_path, recursive  = TRUE)
-
+      dir.create(file.path(pkg,validation_directory,type,item_dir), recursive  = TRUE, showWarnings = FALSE)
```

use `fs::dir_create` to silently ignore existing directories